src/Core/Managed/Shared/TelemetryClient.cs

```diff
@@ -360,7 +365,7 @@ public void Initialize(ITelemetry telemetry)
             var telemetryWithProperties = telemetry as ISupportProperties;
             if (telemetryWithProperties != null)
             {
-                if (this.Channel.DeveloperMode.HasValue && this.Channel.DeveloperMode.Value)
+                if ((this.Channel != null) && (this.Channel.DeveloperMode.HasValue && this.Channel.DeveloperMode.Value))
```

Comment:
Why would we try to enrich data that we will drop? Let's remove this check and throw before we get here